src/oracleFactory/createOracle.ts

```diff
@@ -151,10 +143,10 @@ const createApiKeyDataset = ({
           const order = await iexec.order
             .signDatasetorder(orderToSign)
             .catch((e: Error) => {
-              throw new WorkflowError(
-                "Failed to sign API key datasetorder's",
-                e
-              );
+              throw new WorkflowError({
+                message: "Failed to sign API key datasetorder's",
+                errorCause: e,
+              });
```

Comment:
why keep the catch here and not above for `iexec.order.createDatasetorder` ?

Reply:
need to be update

Reply:
Re-added for createDataset order because neither of these actions can cause a market api call error